paddle/fluid/framework/executor.cc

```diff
@@ -671,8 +671,22 @@ void Executor::EnableMKLDNN(const ProgramDesc& program) {
   for (size_t bid = 0; bid < program.Size(); ++bid) {
     auto* block = const_cast<ProgramDesc&>(program).MutableBlock(bid);
     for (auto* op : block->AllOps()) {
-      if (op->HasAttr("use_mkldnn")) {
-        op->SetAttr("use_mkldnn", true);
+      auto op_type = op->Type();
+
+      auto& all_kernels = OperatorWithKernel::AllOpKernels();
+      auto it = all_kernels.find(op_type);
+      if (it != all_kernels.end())
+        for (auto& kernel_pair : it->second)
+          if (platform::is_cpu_place(kernel_pair.first.place_) &&
+              (kernel_pair.first.library_type_ == LibraryType::kMKLDNN))
+            op->SetAttr("use_mkldnn", true);
```

Comment:
Could you add a `break` after this line to exit the for loop after it is already set to true?
Or even it would be great for readability to put this block in a function that returns a boolean. Something like 
```c++
if (FoundOneDNNKernel(op_type)) op->SetAttr("use_mkldnn", true);
```

Reply:
This is a nice suggestion, the code will be more readable

Reply:
I corrected it a bit, but due to the fact that in the executor I can't get to the data type because we only have OpDesc. In mkldnn_placement_pass we have access to Node, which allows us to check the data type.